test/e2e/kubectl.go

```diff
@@ -97,11 +97,12 @@ var _ = Describe("kubectl", func() {
 	Describe("guestbook", func() {
 		var guestbookPath = filepath.Join(testContext.RepoRoot, "examples/guestbook")
 
-		if testContext.Provider != "gce" && testContext.Provider != "gke" {
-			By(fmt.Sprintf("Skipping guestbook, uses createExternalLoadBalancer, a (gce|gke) feature"))
-		}
-
 		It("should create and stop a working application", func() {
+			if testContext.Provider != "gce" && testContext.Provider != "gke" {
```

Comment:
It isn't a pointer to string, so it can't be nil. I added a check here and everywhere else we use this value to make sure it isn't the empty string (which was many places other than these two files).